DXRVanilla/DeusEx/Classes/AlarmLight.uc

```diff
@@ -21,45 +21,24 @@ simulated function Timer()
         if (p!=None){ //Wait until the player is here
             SetTimer(0,false);
             kludged=True;
-            DancePartyKludge();
+            SpawnDXRando(p);
         }
     }
-
 }
-//Just kludging some Rando logic into the Alarm Lights so we can do some goofy stuff to the dance party ending
-function DancePartyKludge()
-{
-    local DeusExPlayer p;
-    local POVCorpse c;
-    local vector loc;
-    local Rotator rot;
-    local bool broughtLeo;
-    local bool alive;
-    local TerroristCommander leo;
 
-    foreach AllActors(class'DeusExPlayer',p){
-        log("Player has "$p.inHand$" in hands");
-        if (p.inHand.IsA('POVCorpse')){
-            c = POVCorpse(p.InHand);
-            if (c.carcClassString == "DeusEx.TerroristCommanderCarcass"){
-                broughtLeo = true;
-                alive = c.bNotDead;
-            }
-        }
-    }
+function SpawnDXRando(DeusExPlayer player)
+{
+    local DeusExLevelInfo DeusExLevelInfo;
+    local PlayerStart ps;
 
-    if (broughtLeo){
-        loc.X = -736;
-        loc.Y = -22;
-        loc.Z = -32;
+    class'DeusExLevelInfo'.default.MapName = "ENDGAME4";
+    class'DeusExLevelInfo'.default.missionNumber = 99;
+    class'DeusExLevelInfo'.default.Script = class'MissionEndgame';
+    class'DeusExLevelInfo'.default.ConversationPackage="DeusExConversations";
```

Comment:
did we check if there's any reason to put these defaults back to their original values? like maybe if you start a new game after this?

Reply:
I would expect that it wouldn't be an issue, since DeusExLevelInfos don't need to be spawned in any other levels (in theory), and the ones that exist should have their values predefined in the map.

That being said, yeah, it should probably still reset them afterwards just for safety